### PR TITLE
Use `egui::Modal` to implement our modal dialogs

### PR DESCRIPTION
**Pull request by abey79**
### Related

* Based on https://github.com/emilk/egui/pull/5358
* Based on https://github.com/emilk/egui/pull/5414

### What

Our modal used to be implemented using a wrapper which used `egui::Window`. The guts of that wrapper are now reimplemented based on `egui::Modal`. That wrapper is now named `ModalWrapper`. This does not affect "user" code, which can still use the existing `ModalHandler`.

This PR does not affect the styling of the modals. The only difference is modal are now centred on the window (that `egui::Modal`'s default behaviour). Previously they were slightly offset towards the top of the window.

image

### TODO
- [x] add a snapshot test
- [x] test all modals
- [x] update egui refs when https://github.com/emilk/egui/pull/5414 is merged

### CODE REVIEW

File: crates/viewer/re_ui/tests/snapshots/modal_list_item.png
Comment (teh-cmc) [inline crates/viewer/re_ui/tests/snapshots/modal_list_item.png:1]: Am I somehow magically looking at LFS-backed data, or was this committed directly to the repo?

File: crates/viewer/re_ui/tests/snapshots/modal_normal.png
Comment (teh-cmc) [inline crates/viewer/re_ui/tests/snapshots/modal_normal.png:1]: Ditto

**Review by teh-cmc — APPROVED** [review]

File: crates/viewer/re_ui/tests/snapshots/modal_list_item.png
Comment (abey79) [reply crates/viewer/re_ui/tests/snapshots/modal_list_item.png:1]: TL;DR: yes.

GH is extremely bad at displaying that stuff. The best I could do is to go to the relevant branch, navigate to [the file in question](https://github.com/rerun-io/rerun/blob/05a3f7e348b289e20a75a5691f0af1ff286019f6/crates/viewer/re_ui/tests/snapshots/modal_list_item.png), and notice the tiny mention top left:

image

Also, CI allegedly checks for that.

Comment (abey79) [reply crates/viewer/re_ui/tests/snapshots/modal_list_item.png:1]: TIL you can navigate to that file by using the … menu in the diff view and pick "View file".

File: crates/viewer/re_ui/tests/snapshots/modal_normal.png
Comment (abey79) [reply crates/viewer/re_ui/tests/snapshots/modal_normal.png:1]: Ditto

**Pull request merged**